src/main/java/com/getindata/connectors/http/internal/table/lookup/JavaNetHttpPollingClient.java

```diff
@@ -122,4 +135,24 @@ private boolean notErrorCodeAndNotEmptyBody(String body, int statusCode) {
     HttpRequestFactory getRequestFactory() {
         return this.requestFactory;
     }
+
+    private Collection<RowData> deserialize(String responseBody) throws IOException {
+        byte[] rawBytes = responseBody.getBytes();
+        String resultType = options.getProperties().getProperty(RESULT_TYPE, "single-value");
+        if (resultType.equals("single-value")) {
+            return Collections.singletonList(responseBodyDecoder.deserialize(rawBytes));
```

Comment:
Watch out null, according to flink [docs](https://github.com/apache/flink/blob/9a1c987f0d924977e87dada55f1e1e7d84b5e4c9/flink-core/src/main/java/org/apache/flink/api/common/serialization/DeserializationSchema.java#L67) deserializer returns null if message cannot be deserialized